# !/usr/bin/env python

import sys
#sys.path.insert(0, "/Users/crisl-adm/Tools/galaxy/galaxy/tools/eupath/Tools/lib/python")
sys.path.insert(0, "/opt/galaxy/tools/eupath/Tools/lib/python")
from eupath import BigwigCollectionEupathExporter
from eupath import EupathExporter
import optparse
import sys
import re


def main():
    """
      The following program is a Galaxy Tool for exporting bigwig collections from Galaxy to EuPathDB via iRODS.

      It takes the standard exporter params (see EuPathExporter superclass), plus the arguments declared in it
    """

    parser = optparse.OptionParser()
    (options, args) = parser.parse_args()

    # So users are greeted with a bewildering traceback
    sys.tracebacklimit = 0

    exporter = BigwigCollectionEupathExporter.BigwigCollectionExport(args)
    try:
        exporter.export()
    except EupathExporter.ValidationException as ve:
        print >> sys.stderr, str(ve)
        sys.exit(1)


if __name__ == "__main__":
    sys.exit(main())